src/tools/ScenarioMeasurement/Startup/WPFParser.cs

```diff
@@ -7,17 +9,101 @@ public class WPFParser : IParser
+            int? pid = null;
+            bool doneParsingXaml = false;
+            double lastKnownTime = 0.0;
+            using (var source = new ETWTraceEventSource(mergeTraceFile))
+            {
+
+                source.Kernel.ProcessStart += evt =>
+                {
+                    if (processName.Equals(evt.ProcessName, StringComparison.OrdinalIgnoreCase) && pids.Contains(evt.ProcessID) && evt.CommandLine.Trim() == commandLine.Trim())
+                    {
+                        if (pid.HasValue)
+                        {
+                            // Processes might be reentrant. For now this traces the first (outermost) process of a given name.
+                            return;
+                        }
+                        pid = evt.ProcessID;
+                        start = evt.TimeStampRelativeMSec;
+                    }
+                };
+
+                source.Kernel.ThreadCSwitch += evt =>
+                {
+                    if (!pid.HasValue) // we're currently in a measurement interval
+                        return;
+
+                    if (evt.NewProcessID != pid && evt.OldProcessID != pid)
+                        return; // but this isn't our process
+
+                    if (evt.OldProcessID == pid) // this is a switch out from our process
+                    {
+                        if (ins.TryGetValue(evt.OldThreadID, out var value)) // had we ever recorded a switch in for this thread? 
+                        {
+                            threadTime += evt.TimeStampRelativeMSec - value;
+                            ins.Remove(evt.OldThreadID);
+                        }
+                    }
+                    else // this is a switch in to our process
+                    {
+                        ins[evt.NewThreadID] = evt.TimeStampRelativeMSec;
+                    }
+                };
+
+                //We want to find the first Present/Stop after the last XamlParse event
+                source.Dynamic.AddCallbackForProviderEvent("Microsoft-Windows-WPF", "WClientParseXamlBamlInfo", evt =>
+                {
+                    if (pid.HasValue && evt.ProcessID == pid && evt.ProcessName.Equals(processName, StringComparison.OrdinalIgnoreCase))
+                    {
+                        doneParsingXaml = true; //So when we find a Xaml parse event set the flag
+                    }
+                });
+
+                source.Dynamic.AddCallbackForProviderEvent("Microsoft-Windows-WPF", "WClientUcePresent/Stop", evt =>
+                {
+                    //Then when we see a Present/Stop after the flag has been set
+                    if (doneParsingXaml && pid.HasValue && evt.ProcessID == pid && evt.ProcessName.Equals(processName, StringComparison.OrdinalIgnoreCase))
+                    {
+                        lastKnownTime = evt.TimeStampRelativeMSec; //Grab the current time
+                        doneParsingXaml = false; //And reset the flag
```

Comment:
It does not appear that we are using this anywhere else, am I missing something?